environs/simplestreams/simplestreams.go

```diff
@@ -1040,6 +1046,11 @@ func UserPublicSigningKey() (string, error) {
 	signingKeyFile := filepath.Join(agent.DefaultConfDir, SimplestreamsPublicKeyFile)
 	b, err := ioutil.ReadFile(signingKeyFile)
 	if os.IsNotExist(err) {
+		logger.Errorf("reading file %v %v", signingKeyFile, err)
+		// After long discussions and debate, it has been deemed
```

Comment:
I didn't say that ;-p.  I still think it's a bad idea and should be discouraged (and I even thought I convinced wallyworld of that in irc); I would prefer it if the comment made it clear that it's bad practice allowed to stand because there are more important things to worry about right no, rather than a legitimate technique.

<fwereade> I understand that -- but that doesn't make "" a valid key, and that makes the func misleading
<fwereade> the client *has* to know that detail
<fwereade> and that's only safe when you're the only person refactoring that code under pressure ;p
<wallyworld> fair point